### PR TITLE
Add log to track field name when exception thrown; Hot fix for calculating buffer length

### PR DESCRIPTION
**Pull request by jackjlli**
This PR adds log to track the field name when exception happens. 
Currently if some exceptions like `IndexOutOfBoundsException` are thrown, we don't know which field it is, which makes us hard to debug.
This PR also hot fixes the calculation of buffer length. Basically if two factors in the multiplication are of type integer, then the result is of type integer as well. Casting one of the factor to long will fix the problem.

### CODE REVIEW

File: pinot-core/src/main/java/com/linkedin/pinot/core/segment/creator/impl/SegmentColumnarIndexCreator.java [hunk 1]
```diff
@@ -133,7 +137,12 @@ public void init(SegmentGeneratorConfig segmentCreationSpec, SegmentIndexCreatio
         _dictionaryCreatorMap.put(columnName, dictionaryCreator);
 
         // Create dictionary
-        dictionaryCreator.build();
+        try {
+          dictionaryCreator.build();
+        } catch (IOException e) {
```
Comment (sunithabeeram) [inline pinot-core/src/main/java/com/linkedin/pinot/core/segment/creator/impl/SegmentColumnarIndexCreator.java]: The PR description says you want to track IndexOutOfBoundsException whereas you are catching IOE and these are thrown from PinotDataBuffer in the build path. You will want to catch Exception in general and rethrow (as currently done).

**Comment by sunithabeeram** [comment]
LGTM. 

I am not sure about the usage of rethrowException in general. Its not clear to my why we can't use throw(e) again after catching it. Will check with @jfim about it.

The changes also included a few formatting only ones - we should keep formatting changes separate from actual code changes. If not already set, we should have the formatter only reformat updated lines - I think you can specify this in intellij as well.

File: pinot-core/src/main/java/com/linkedin/pinot/core/segment/creator/impl/SegmentColumnarIndexCreator.java [hunk 2]
```diff
@@ -133,7 +136,13 @@ public void init(SegmentGeneratorConfig segmentCreationSpec, SegmentIndexCreatio
         _dictionaryCreatorMap.put(columnName, dictionaryCreator);
 
         // Create dictionary
-        dictionaryCreator.build();
+        try {
+          dictionaryCreator.build();
+        } catch (Exception e) {
+          LOGGER.error("Error building dictionary for field: {}, cardinality: {}, max length in bytes: {}",
+              fieldSpec.getName(), indexCreationInfo.getDistinctValueCount(), dictionaryCreator.getNumBytesPerEntry());
+          Utils.rethrowException(e);
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/com/linkedin/pinot/core/segment/creator/impl/SegmentColumnarIndexCreator.java]: Change this to "throw e;"

File: pinot-core/src/main/java/com/linkedin/pinot/core/segment/creator/impl/SegmentColumnarIndexCreator.java [hunk 3]
```diff
@@ -133,7 +136,13 @@ public void init(SegmentGeneratorConfig segmentCreationSpec, SegmentIndexCreatio
         _dictionaryCreatorMap.put(columnName, dictionaryCreator);
 
         // Create dictionary
-        dictionaryCreator.build();
+        try {
+          dictionaryCreator.build();
+        } catch (Exception e) {
+          LOGGER.error("Error building dictionary for field: {}, cardinality: {}, max length in bytes: {}",
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/com/linkedin/pinot/core/segment/creator/impl/SegmentColumnarIndexCreator.java]: Prefer "number bytes per entry" over "max length in bytes"

**Pull request merged**